hoot-core/src/main/cpp/hoot/core/conflate/highway/HighwaySnapMerger.cpp

```diff
@@ -334,7 +372,26 @@ bool HighwaySnapMerger::_mergePair(const OsmMapPtr& map, ElementId eid1, Element
   // remove the old way that was split and snapped
   if (e1 != e1Match && scraps1)
   {
-    ReplaceElementOp(eid1, scraps1->getElementId(), true).apply(result);
+    if (swapWayIds)
+    {
+      ElementId eidm1 = e1Match->getElementId();
+      //  Swap the old way ID back into the match element
+      IdSwapOp(eid1, eidm1).apply(result);
+      //  Remove the old way with a new swapped out ID
+      RemoveElementByEid(eidm1).apply(result);
+      //  Add the scraps element to all the relations that the match is in
+      if (scraps1)
+      {
+        QList<ElementPtr> list;
+        list.append(e1Match);
+        list.append(scraps1);
+        result->replace(e1Match, list);
+        //  Update the scraps
+        _updateScrapParent(result, e1Match->getId(), scraps1);
+      }
```

Comment:
Swap the way IDs  of original way that is going to be removed to the `e1Match` element, preserving the original ID.